File: ios/swiftHTMLWebviewApp/Models/BridgeDispatcher.swift
//
//  BridgeDispatcher.swift
//  swiftHTMLWebviewApp
//

import Foundation

enum BridgeDispatcher {
    static func action(from request: [String: Any]) -> String? {
        guard let action = request["action"] as? String else {
            return nil
        }
        let trimmed = action.trimmingCharacters(in: .whitespacesAndNewlines)
        return trimmed.isEmpty ? nil : trimmed
    }

    static func missingActionResponse(request: [String: Any], message: String) -> [String: Any] {
        BridgeResponse.error(request: request, action: "unknown", message: message)
    }

    static func unknownActionResponse(request: [String: Any], action: String, message: String) -> [String: Any] {
        BridgeResponse.error(request: request, action: action, message: message)
    }
}

File: android/app/src/main/java/com/ilass/swifthtmlwebviewapp/BridgeDispatcher.java
package com.ilass.swifthtmlwebviewapp;

import org.json.JSONException;
import org.json.JSONObject;

final class BridgeDispatcher {
    private BridgeDispatcher() {
    }

    static String action(JSONObject message) {
        if (message == null) {
            return "";
        }
        return message.optString("action", "").trim();
    }

    static JSONObject missingActionResponse(JSONObject message) throws JSONException {
        return BridgeResponse.error(message, "unknown", "Invalid request: Missing 'action' parameter.");
    }

    static JSONObject unknownActionResponse(JSONObject message, String action) throws JSONException {
        return BridgeResponse.error(message, action, "Unknown native action: " + action);
    }

    static JSONObject parseErrorResponse(String error) throws JSONException {
        return BridgeResponse.error(null, "unknown", error);
    }
}
